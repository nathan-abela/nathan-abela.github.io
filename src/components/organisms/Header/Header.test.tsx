import { render } from '@testing-library/react';

import { Header } from './Header';

const mockDispatch = jest.fn();

jest.mock('react-redux', () => ({
	useSelector: jest.fn(),
	useDispatch: () => mockDispatch,
}));

describe('Header', () => {
	it('renders correctly', () => {
		const { container } = render(
			// eslint-disable-next-line prettier/prettier
			<Header />
		);
		expect(container).toMatchSnapshot();
	});
});
