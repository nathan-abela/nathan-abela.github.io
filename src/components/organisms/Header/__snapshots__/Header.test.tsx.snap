// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div>
  <header>
    <div
      class="styles__StyledContainer-sc-a6yh8x-0 earCTD"
    >
      <div
        class="styles__Flex-sc-1w2me4-0 bHDEaA"
      >
        <h1>
          <a
            href="/"
          >
            Nathan Abela
          </a>
        </h1>
        <div
          class="styles__Desktop-sc-1w2me4-1 kRNKOm"
        >
          <nav
            class="styles__StyledNavigation-sc-1f4kyre-0"
          >
            <ul
              class="styles__List-sc-1f4kyre-1 ivSflr"
            >
              <li>
                <a
                  href="/"
                >
                  Home
                </a>
              </li>
              <li>
                <a
                  href="/about"
                >
                  About
                </a>
              </li>
              <li>
                <a
                  href="/education"
                >
                  Education
                </a>
              </li>
              <li>
                <a
                  href="/work"
                >
                  Work
                </a>
              </li>
              <li>
                <a
                  href="/portfolio"
                >
                  Portfolio
                </a>
              </li>
              <li
                class="desktop-only"
              >
                <button
                  class="styles__StyledThemeToggle-sc-1b7mai8-0 kHwexB"
                  title="Switch to light theme"
                >
                  <test-file-stub />
                </button>
              </li>
            </ul>
          </nav>
        </div>
        <div
          class="styles__Mobile-sc-1w2me4-2 kpxcyD"
        >
          <ul
            class="styles__StyledMobileNavigation-sc-1619527-0 huyxbx"
          >
            <li>
              <button
                class="styles__StyledThemeToggle-sc-1b7mai8-0 kHwexB"
                title="Switch to light theme"
              >
                <test-file-stub />
              </button>
            </li>
            <li>
              <button
                class="styles__Menu-sc-1619527-1 hKtafq"
              >
                <test-file-stub />
              </button>
            </li>
          </ul>
          <div
            class="styles__Overlay-sc-1619527-2 lnRURg"
          >
            <nav
              class="styles__StyledNavigation-sc-1f4kyre-0"
            >
              <ul
                class="styles__List-sc-1f4kyre-1 ivSflr"
              >
                <li>
                  <a
                    href="/"
                  >
                    Home
                  </a>
                </li>
                <li>
                  <a
                    href="/about"
                  >
                    About
                  </a>
                </li>
                <li>
                  <a
                    href="/education"
                  >
                    Education
                  </a>
                </li>
                <li>
                  <a
                    href="/work"
                  >
                    Work
                  </a>
                </li>
                <li>
                  <a
                    href="/portfolio"
                  >
                    Portfolio
                  </a>
                </li>
                <li
                  class="desktop-only"
                >
                  <button
                    class="styles__StyledThemeToggle-sc-1b7mai8-0 kHwexB"
                    title="Switch to light theme"
                  >
                    <test-file-stub />
                  </button>
                </li>
              </ul>
            </nav>
          </div>
        </div>
      </div>
    </div>
  </header>
</div>
`;
